eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -551,8 +555,8 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: A
   }
 
   def createNewChannel(nodeParams: NodeParams, funder: Boolean, fundingAmount: Satoshi, origin_opt: Option[ActorRef]): (ActorRef, LocalParams) = {
-    val defaultFinalScriptPubKey = Helpers.getFinalScriptPubKey(wallet, nodeParams.chainHash)
-    val localParams = makeChannelParams(nodeParams, defaultFinalScriptPubKey, funder, fundingAmount)
+    val localPaymentPubkey = Helpers.getWalletPaymentBasepoint(wallet)
+    val localParams = makeChannelParams(nodeParams, localPaymentPubkey, funder, fundingAmount)
```

Comment:
With this change, our final pubkeyscript will always be a 'p2wpkh' script even if we don't set `static_remote_key`. This will break eclair-mobile which only understands `p2sh-of-p2wpkh` for now.

Reply:
This brings up another point as well, the PR is removing the `claimP2WPKHOutputTx` transaction because our `toRemote` output of the remote commit is a key from the BIP32 wallet and we don't need to 'spend back'. However eclair-mobile doesn't currently understands `p2wpkh` outputs so it won't be able to spend out `toRemote` output. Possible solutions:
- improve eclair-mobile to understand `p2wpkh` outputs
- do the `claimP2WPKHOutputTx` to spend back to a `p2sh-p2wpkh`

Reply:
We will add support for `p2wpkh` to eclair-mobile, but for now the simplest solution is to not change how it works

Reply:
Latest commit d956374 is reverting the logic for the creation of the finalScriptPubkey for channels that don't support `option_static_remotekey`